### PR TITLE
[docs] [2.20.1, 2.18.5] added editable fields for multiple cloud providers 

### PR DESCRIPTION
**Pull request by aishwarya24**
related to DOC-191

### CODE REVIEW

**Comment by Jethro-M** [comment]
Looks good!
One more thing to add is a note about region and zone editing.

Users may add regions and zones to an in-use provider.
Users are unable to edit region details (ex. shared subnet, yb image etc.), or delete region if any of the region's zones are in use.
Users are unable to delete zones which are in-use.

**Review by Jethro-M — APPROVED** [review]

File: docs/content/stable/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/gcp.md [hunk 1]
```diff
@@ -128,7 +128,23 @@ Navigate to **Configs > Infrastructure > Google Cloud Platform** to see a list o
 
 To view a provider, select it in the list of GCP Configs to display the **Overview**.
 
-To edit the provider, select **Config Details**, make changes, and click **Apply Changes**. For more information, refer to [Provider settings](#provider-settings). Note that, depending on whether the provider has been used to create a universe, you can only edit a subset of options.
+To edit the provider, select **Config Details**, make changes, and click **Apply Changes**. For more information, refer to [Provider settings](#provider-settings). Note that for YBA version 2.20.1 and later, depending on whether the provider has been used to create a universe, you can only edit a subset of fields such as the following:
+
+- Provider Name
+- Credential Type. You can upload a new Google Service Account JSON file (`gceApplicationCredentials`). Note that the `project_id` field in the file should not have a new entry. If the file contains a new GCE project,
```
Comment (ddhodge) [inline docs/content/stable/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/gcp.md:134]:
```suggestion
- Credential Type. You can upload a new Google Service Account JSON file (`gceApplicationCredentials`). Note that the `project_id` field can't have a new entry. For example:
```

File: docs/content/stable/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/gcp.md [hunk 2]
```diff
@@ -128,7 +128,23 @@ Navigate to **Configs > Infrastructure > Google Cloud Platform** to see a list o
...
+
+    ```json
+    {
+      "type": "service_account",
+      "project_id": "new-project-yb",
+       ...
+    }
+    ```
+
+    then the backend request fails and you will receive a toast notification via the UI indicating that you can't change the GCE project for an in-use provider.
```
Comment (ddhodge) [inline docs/content/stable/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/gcp.md:144]:
```suggestion
    If `new-project-yb` is a new GCE project, the backend request fails and you will be notified that you can't change the GCE project for an in-use provider.
```

File: docs/content/stable/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/aws.md [hunk 3]
```diff
@@ -95,7 +95,13 @@ Navigate to **Configs > Infrastructure > Amazon Web Services** to see a list of
 
 To view a provider, select it in the list of AWS Configs to display the **Overview**.
 
-To edit the provider, select **Config Details**, make changes, and click **Apply Changes**. For more information, refer to [Provider settings](#provider-settings). Note that, depending on whether the provider has been used to create a universe, you can only edit a subset of options.
+To edit the provider, select **Config Details**, make changes, and click **Apply Changes**. For more information, refer to [Provider settings](#provider-settings). Note that for YBA version 2.20.1 and later, depending on whether the provider has been used to create a universe, you can only edit a subset of fields such as the following:
```
Comment (ddhodge) [inline docs/content/stable/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/aws.md:98]:
```suggestion
To edit the provider, select **Config Details**, make changes, and click **Apply Changes**. For more information, refer to [Provider settings](#provider-settings). Note that for YBA version 2.20.1 and later, if the provider has been used to create a universe, you can only edit a subset of fields, including the following:
```

File: docs/content/stable/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/kubernetes.md [hunk 4]
```diff
@@ -194,7 +194,13 @@ Navigate to **Configs > Infrastructure > Managed Kubernetes Service** to see a l
 
 To view a provider, select it in the list of Managed Kubernetes Service Configs to display the **Overview**.
 
-To edit the provider, select **Config Details**, make changes, and click **Apply Changes**. For more information, refer to [Provider settings](#provider-settings). Note that, depending on whether the provider has been used to create a universe, you can only edit a subset of options.
+To edit the provider, select **Config Details**, make changes, and click **Apply Changes**. For more information, refer to [Provider settings](#provider-settings). Note that for YBA version 2.20.1 and later, depending on whether the provider has been used to create a universe, you can only edit a subset  of fields such as the following:
+
+- Provider Name
+- Pull Secret
+- Kube Config (kubeConfigName, kubeConfigContent, kubeConfig fields in the Kube Config File)
```
Comment (ddhodge) [inline docs/content/stable/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/kubernetes.md:201]:
```suggestion
- Kube Config (kubeConfigName, kubeConfigContent, kubeConfig fields in the Kube Config file)
```

**Review by Vars-07 — APPROVED** [review]
LGTM!

File: docs/content/stable/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/aws.md [hunk 5]
```diff
@@ -95,7 +95,13 @@ Navigate to **Configs > Infrastructure > Amazon Web Services** to see a list of
...
+
+- Provider Name
+- Access Key ID
+- Secret Access Key
+- Regions - You can add regions and zones to an in-use provider.
+Note that you cannot edit existing region details (such as shared subnet, yb image, and so on), or delete a region if any of the region's zones are in use. You also cannot delete zones which are in-use.
```
[the remark this answers is not in the document]
Comment (aishwarya24) [reply docs/content/stable/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/aws.md]: After more clarification with @Jethro-M , I agree that we don't need to call out the examples at all as the fields are anyway not editable.

**Pull request merged**